Integrations/integration-ProtectWise.yml

```diff
@@ -201,101 +187,300 @@ script:
+                            skip = false;
+                            break;
+                        }
+                    }
+                } else {
+                    skip = false;
+                }
+                if(!skip) {
+                    incidents.push(createIncidentFromEvent(res[i]));
+                }
+            }
+            setLastRun({startTime: startTime});
+            return JSON.stringify(incidents);
         case 'sensors':
-            var token = getToken();
-            return {Type: 1, Contents: sensors(args.sensorId,token), ContentsFormat: 'json'};
+        case 'protectwise-show-sensors':
+            var url = params.url+'sensors';
+            if (args.sensorId && args.sensorId.length > 0 ) {
+                url = url + '/' + args.sensorId;
+            }
+            var headers = args.headers ? args.headers : DEFAULT_SENSOR_HEADERS;
+
+
+            var res = doReq(url, token);
+            var items = [];
+
+            if (! Array.isArray(res)) {
+                res = [res];
+            }
+            items = filterClone(res, headers);
+
+            return {
+                Type: entryTypes.note,
+                Contents: res,
+                ContentsFormat: formats.json,
+                HumanReadable: tableToMarkdown('Protectwise sensors', items, headers),
+                EntryContext: {'Protectwise.Sensor(val.id == obj.id)': items}
+            };
         case 'search':
-            var token = getToken();
-            return {Type: 1, Contents: eventsSearch(args.start,args.end,args.eventType,args.killChainStage,args.threatLevel,
-                args.threatCategory,args.observationStage,args.ip,args.expandDetails,args.minLimit,args.maxLimit,args.reverseOrder,args.nextPage,token), ContentsFormat: 'json'};
+        case 'protectwise-search-events':
+            var raw = eventsSearch(args.start,args.end,args.eventType,args.killChainStage,args.threatLevel,
+                args.threatCategory,args.observationStage,args.ip,args.expandDetails,args.minLimit,args.maxLimit,args.reverseOrder,args.nextPage,token);
+            var res = JSON.parse(raw).events;
+            var eventHeaders = args.headers ? args.headers : DEFAULT_EVENT_HEADERS;
+            var items = [];
+            if (! Array.isArray(res)) {
+                res = [res];
+            }
+            items = filterClone(res, eventHeaders);
+            items.forEach(function(item) {
+               TIME_FIELDS.forEach(function (timeKey) {
+                   if (item[timeKey]) {
+                       item[timeKey] = convertTimestampToString(item[timeKey]);
+                   }
+               });
+            });
+
+            return {
+                Type: entryTypes.note,
+                Contents: raw,
+                ContentsFormat: formats.json,
+                HumanReadable: tableToMarkdown('Protectwise Event Search', items, eventHeaders),
+                EntryContext: {'Protectwise.Event(val.id == obj.id)': items}
```

Comment:
Right.. it's an array of events. But it merges on a per-item basis
Seemed to work at least when I tested, let me know if you think I am misunderstanding